apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx

```diff
@@ -85,63 +91,66 @@ const SignUpForm = ({ userData, setUserData, onSignUp }: SignUpFormProps) => {
     event.preventDefault();
   };
 
-  const emailValidators: ValidatorFunction[] = [
-    (value: string) => ({
-      valid: emailMatcher.test(value),
-      error: shared.emailFieldError,
-    }),
-  ];
-
-  const validatePassword = useCallback(() => {
-    setIsPasswordValid(passwordMatcher.test(password));
-  }, [password]);
-
-  useEffect(() => {
-    validatePassword();
-  }, [password, validatePassword]);
-
-  const onSignUpClick = useCallback(() => {
+  const validateForm = useCallback(() => {
     const { username, confirmPassword, firstName, lastName, password } =
       userData;
 
-    if (
-      [firstName, lastName, username, password, confirmPassword].some(
-        (value) => !value || value.trim().length === 0
-      )
-    ) {
-      return;
-    }
-
-    if (password !== confirmPassword) {
-      return;
-    }
+    const nameError = !firstName || firstName.trim().length === 0;
+    const surnameError = !lastName || lastName.trim().length === 0;
+    const emailError =
+      !username || username.trim().length === 0
```

Comment:
This logic is duplicate from apps/nextjs-website/src/components/organisms/Auth/LoginForm.tsx, I'd put it in a separate util function